### DISCUSSION TITLE
Symfony UX Turbo-Stream update on link (a-tag) click

### DISCUSSION THREAD

**Discussion by PhilippBartels** [symfony/symfony#43335] [Q&A] [answered]
Hey people,

I've got a symfony application and now I'm implementing Symfony's UX-Turbo bundle. I already changed my form handling and the functionality, including live update with streaming, is working like a charm. But I'm struggeling with the implementation of delete links using a simple a-tag with href.

I created a route for the deletion of the specific database object and after that the users is redirected to the edit-page again. 
I already added this in my controller so it should answer with a streaming-template if the request is correct:

 ```
   if (TurboStreamResponse::STREAM_FORMAT === $request->getPreferredFormat()) {
        return $this->render('client/list_types/_table_association_body.stream.html.twig', ["client" => $clientUserAssociation->getClient()], new TurboStreamResponse());
    }
```

But every time i click on the delete-link the browser behaviour is just normal. No ajax request or something. The page is loading and then the redirections happens. Is there a way to force the link to behave like a form submit or something?

If not i have to implement a Form with the specific option selected in a hidden field and only show the delete button, which in fact would be a form-submit-button. But this feels like a dirty workaround. I'm sure there is a solution for this.

Here is my related SO post: https://stackoverflow.com/questions/69460656/symfony-ux-turbo-fetch-link-in-background-and-update

Thanks for your help!

**Comment by stof** [comment, marked as answer]
> If not i have to implement a Form with the specific option selected in a hidden field and only show the delete button, which in fact would be a form-submit-button. But this feels like a dirty workaround. I'm sure there is a solution for this.

this is *not* a dirty workaround. This is also the way you would implement a delete button without Turbo if you respect the HTTP semantic, as deleting something must not be done with a GET request.

**Reply by PhilippBartels** [reply]
Hey, thanks for your answer. I implemented it with a form, and it's working fine. I'm closing this now.
